pkg/sparrow/targets/targetmanager.go

```diff
@@ -81,5 +84,12 @@ func (c *TargetManagerConfig) Validate(ctx context.Context) error {
 		log.Error("The update interval should be equal or above 0", "interval", c.UpdateInterval)
 		return ErrInvalidUpdateInterval
 	}
-	return nil
+
+	switch c.Type {
+	case "gitlab":
```

Comment:
I don't like that we have to write out the type here and in pkg/sparrow/targets/interactor/config.go ...

One suggestion would be to define the `Config` as a map and not a struct in pkg/sparrow/targets/interactor/config.go

Reply:
I've started to implement your proposal:
```go
package interactor
// Config contains the configuration for the remote interactor
type Config map[Type]struct {
	// Gitlab contains the configuration for the gitlab interactor
	Gitlab gitlab.Config `yaml:"gitlab" mapstructure:"gitlab"`
}

const GitLab Type = "gitlab"

func (t Type) Interactor(cfg Config) remote.Interactor {
	c := cfg[t]
	switch t {
	case GitLab:
		return gitlab.New(c.Gitlab)
	default:
		return nil
	}
}
```
It would indeed genericize the validation:
```go
	if _, ok := c.Config[c.Type]; !ok {
		log.ErrorContext(ctx, "Invalid interactor type", "type", c.Type)
		return ErrInvalidInteractorType
	}
```

But idk about this approach. It gives some flexibility and simplicity to add new interactors but we could also introduce less type-safety and it might be less intuitive to use. Also I haven't tried if the map is squashed correctly into the target manager config.

I'd prefer to just add constants for each interactor type and then use them in the Validate method but let me know what you think.

Reply:
I've implemented the gitlab type constant as discussed in our call earlier.